src/structured-data/structured-data-tags.js

```diff
@@ -254,9 +254,24 @@ function generateLiveBlogPostingData(structuredData = {}, story = {}, publisherC
   };
 }
 
+function getEmbedUrl({cards}) {
```

Comment:
use lodash get to get cards, fallback to empty array. Use reduce instead of for loop

Reply:
Using lodash to get cards. The main reason why I used normal loops is to achieve an early return and not iterate every element

Reply:
```js
cards
    .reduce((acc, card) => {
      acc.push(...card["story-elements"]);
      return acc;
    }, [])
    .filter((storyElement) => {
      // filter logic
    });
```

Reply:
That's the issue I'm trying to prevent actually. The filter is run for every element whereas with a normal loop it's easy to exit early. The filter method is fine for browsers but for servers wouldn't you prefer a more performant one? If performance is not an issue, I can go with your approach

Reply:
@ags1773 Using flat map instead of reduce because that's more ideal for this use case

Reply:
can use `find` on the first level of iteration which suffices both the conditions